src/sweetalert2.js

```diff
@@ -597,6 +597,20 @@ const sweetAlert = (...args) => {
       } else {
         error('Unexpected type of inputOptions! Expected object, Map or Promise, got ' + typeof params.inputOptions)
       }
+    } else if (['text', 'email', 'number', 'tel', 'textarea'].includes(params.input) && params.inputValue instanceof Promise) {
+      sweetAlert.showLoading()
+      dom.hide(input)
+      params.inputValue.then((inputValue) => {
+        input.value = inputValue + ''
+        dom.show(input)
+        sweetAlert.hideLoading()
+      })
+      .catch((err) => {
+        error('Error in inputValue promise: ' + err)
+        input.value = ''
```

Comment:
Why do we need to set input value to empty string in case of promise was errorred? Should we just do nothing to input in this case?

Reply:
The reason is that input.value is assigned to params.inputValue in https://github.com/gverni/sweetalert2/blob/224fd940c6255a181a686e3f8945827a38e75623/src/sweetalert2.js#L489 

In case this is a promise and the promise is rejected, the input value will keep the original value the user will see the value of input as "[object Promise]". 

We could workaround that avoiding to assign the input.value in case this is a promise. Would you want me to work on that?

Reply:
thanks for the explanation, no actions needed here. Let's keep it like it is now :+1: